src/main/kotlin/org/rust/ide/navigation/goto/RsGeneratedSourcesFilter.kt

```diff
@@ -8,16 +8,22 @@ package org.rust.ide.navigation.goto
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.roots.GeneratedSourcesFilter
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiDirectory
 import com.intellij.psi.PsiElement
+import org.rust.lang.RsLanguage
 import org.rust.lang.core.macros.findNavigationTargetIfMacroExpansion
 import org.rust.lang.core.macros.macroExpansionManager
+import org.rust.lang.core.psi.isNotRustFile
 
 class RsGeneratedSourcesFilter : GeneratedSourcesFilter() {
     override fun isGeneratedSource(file: VirtualFile, project: Project): Boolean {
+        if (file.isNotRustFile) return false
```

Comment:
Looks like it doesn't allow marking directories as excluded
It works as before for macro expansion files (btw why? maybe we want to mark all files including directories inside `rust_expanded_macros` as generated?) but in #4542 I want to mark out dir as generated as well

Reply:
Ook, this is not related to the issue, so I can remove this line. I just want to prevent access to MacroExpansionManager in the case of non-rust project